cc/imgproc/imgprocBindings.h

```diff
@@ -269,6 +269,124 @@ namespace ImgprocBindings {
 			};
 		};
 	};
+
+  typedef struct HistAxes {
+    float range[2];
+    int channel;
+    int bins;
+  } HistAxes;
+
+  class HistAxesConverterImpl : public FF::UnwrapperBase<HistAxesConverterImpl, HistAxes> {
+  public:
+
+    typedef HistAxes Type;
+
+    static std::string getTypeName() {
+      return std::string("HistAxes");
+    }
+
+    static bool assertType(v8::Local<v8::Value> jsVal) {
+      if (!jsVal->IsObject()) return false;
+
+      auto jsObj = Nan::To<v8::Object>(jsVal).ToLocalChecked();
+
+      if (!FF::hasOwnProperty(jsObj, "ranges")) return false;
+
+      auto jsRangesVal = Nan::Get(jsObj, Nan::New("ranges").ToLocalChecked()).ToLocalChecked();
+      if (!jsRangesVal->IsArray()) return false;
+      auto jsRanges = v8::Local<v8::Array>::Cast(jsRangesVal);
+      if (jsRanges->Length() != 2) return false;
+
+      return (
+          Nan::Get(jsObj, Nan::New("channel").ToLocalChecked()).ToLocalChecked()->IsNumber() ||
+          Nan::Get(jsObj, Nan::New("bins").ToLocalChecked()).ToLocalChecked()->IsNumber()
+      );
+    }
+
+    static HistAxes unwrapUnchecked(v8::Local<v8::Value> jsVal) {
+      FF::TryCatch tryCatch("Imgproc::CalcHist");
+      HistAxes ret;
+      auto jsAxis = Nan::To<v8::Object>(jsVal).ToLocalChecked();
+
+      v8::Local<v8::Value> jsRangesVal = Nan::Get(jsAxis, Nan::New("ranges").ToLocalChecked()).ToLocalChecked();
+      v8::Local<v8::Array> jsRanges = v8::Local<v8::Array>::Cast(jsRangesVal);
+
+      ret.range[0] = FF::DoubleConverter::unwrapUnchecked(Nan::Get(jsRanges, 0).ToLocalChecked());
+      ret.range[1] = FF::DoubleConverter::unwrapUnchecked(Nan::Get(jsRanges, 1).ToLocalChecked());
+      FF::IntConverter::prop(&ret.channel, "channel", jsAxis);
+      FF::IntConverter::prop(&ret.bins, "bins", jsAxis);
+      return ret;
+    }
+
+    static v8::Local<v8::Value> wrap(HistAxes val) {
+      v8::Local<v8::Object> ret = Nan::New<v8::Object>();
+      Nan::Set(ret, FF::newString("bins"), FF::IntConverter::wrap(val.bins));
+      Nan::Set(ret, FF::newString("channel"), FF::IntConverter::wrap(val.channel));
+      Nan::Set(ret, FF::newString("ranges"), FF::FloatArrayConverter::wrap({val.range[0], val.range[1]}));
+
+      return ret;
+    }
+  };
+  typedef FF::ArrayConverterTemplate<HistAxesConverterImpl> ArrayHistAxesConverter;
```

Comment:
What if we isntead expose a HistAxes custom class? This way it would be more convenient to create in javascript instead of having to pass jsons around and we don't have to implement a custom converter.

I am thinking of something like this:
``` c++
namespace internal {
  class HistAxes {
    float range[2];
    int channel;
    int bins;
  }
}

class HistAxes: public FF::ObjectWrap<HistAxes, internal::HistAxes> {}
```

Defining a set of internal helper classes could be useful, if the ObjectWrap class holds multiple native values. Our custom Contours class is another example, which should actually wrap "points" and "hierarchy", but with the current approach only wraps the points.

What do you think?

Reply:
I did not want to introduce breaking changes, that's why I mainly went this route. 
But your approach looks less flaky and less cluttered. I'll give it a try and see how it shapes up